### ISSUE TITLE
Allow refresh to be registered for multiple key-values with same key but different labels

### ISSUE THREAD

**Issue by abhilasharora** [Azure/AppConfiguration-DotnetProvider#156]
This issue tracks the changes needed to allow users to register refresh for multiple key-values with the same key but different labels.

```` csharp
var configBuilder = new ConfigurationBuilder().AddAzureAppConfiguration(options =>
{
    options.Connect(connectionString)
        .Select(KeyFilter.Any, LabelFilter.Null)
        .Select(KeyFilter.Any, "some_label")
        .ConfigureRefresh(refresh =>
        {
            refresh.Register("some_key", LabelFilter.Null, true)
                   .Register("some_key", "some_label", true);
        })
});

**Comment by avanigupta** [comment]
## Proposed Solution:

Consider the following example to better understand the solution:

Let's say that your App Configuration has the following key-values:

Key | Label | Value
--- |--- |---
TestKey | Label1 | Value1
TestKey | Label2 | Value2

### Existing behavior:

Key-values registered for refresh overwrite key-values loaded as part of `Select` clause(s). This means that if some key-value is loaded by `Select` clause and the same key (with a different label) is registered for refresh, your application configuration will always contain the value corresponding to the refresh registered key-value. 

```cs
var config = new ConfigurationBuilder()
    .AddAzureAppConfiguration(options =>
    {
        options.Select(KeyFilter.Any, "Label1");
        options.Select(KeyFilter.Any, "Label2");
        options.ConfigureRefresh(refreshOptions =>
        {
            refreshOptions.Register("TestKey", "Label1")
                          .SetCacheExpiration(TimeSpan.FromSeconds(10));
        });

        refresher = options.GetRefresher();
    })
    .Build();
```

Your application configuration will have the following key-value: 

Key | Value
--- |---
TestKey |  Value1


### New functionality when `refreshAll` is `true`:

Maintaining the behavior that currently exists, refresh registered key-values overwrite key-values loaded by the `Select` clause(s). In addition, if there are multiple refresh registrations of same key with different labels, only the last key-value registered for refresh will be added to your application configuration. However, the refresh registered sentinel keys which are not added to application configuration can still be used to trigger a `refreshAll` operation, in case the sentinel key-value changes.

```cs
var config = new ConfigurationBuilder()
    .AddAzureAppConfiguration(options =>
    {
        options.Select(KeyFilter.Any, "Label1");
        options.Select(KeyFilter.Any, "Label2");
        options.ConfigureRefresh(refreshOptions =>
        {
            refreshOptions.Register("TestKey", "Label1", refreshAll: true)
                          .Register("TestKey", "Label2", refreshAll: true)
                          .SetCacheExpiration(TimeSpan.FromSeconds(10));
        });

        refresher = options.GetRefresher();
    })
    .Build();
```

After initial configuration load completes, your application configuration will have the following key-value: 

Key | Value
--- |---
TestKey |  Value2

Note that even though `Value1` is not in your application configuration, any changes to `TestKey` with `Label1` will still force the entire configuration to update.


### New functionality when `refreshAll` is `false`:

If same key with multiple labels is registered for refresh, and `refreshAll` is not `true`, your application configuration will contain the value corresponding to the key-value most recently fetched from server.

If your App Configuration has the following key-values:

```cs
var config = new ConfigurationBuilder()
    .AddAzureAppConfiguration(options =>
    {
        options.Select(KeyFilter.Any, "Label1");
        options.Select(KeyFilter.Any, "Label2");
        options.ConfigureRefresh(refreshOptions =>
        {
            refreshOptions.Register("TestKey", "Label1")
                          .Register("TestKey", "Label2")
                          .SetCacheExpiration(TimeSpan.FromSeconds(10));
        });

        refresher = options.GetRefresher();
    })
    .Build();
```

After initial configuration load completes, your application configuration will have the following key-value: 

Key | Value
--- |---
TestKey |  Value2

After this, your application configuration will change based on both refresh registered key-values, i.e., if either of the two refresh registered key-values changes in App Configuration, the last changed key-value will be populated in your application configuration. On the next call to `RefreshAsync` or `TryRefreshAsync`, if the cache has expired and the key-value with label `Label1` changes, the new value of `TestKey` with `Label1` will overwrite `Value2` in your application configuration.

**Comment by avanigupta** [comment]
This feature has been released in package version 4.1.0:
- [Microsoft.Extensions.Configuration.AzureAppConfiguration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.AzureAppConfiguration/)
- [Microsoft.Azure.AppConfiguration.AspNetCore](https://www.nuget.org/packages/Microsoft.Azure.AppConfiguration.AspNetCore/)
